app/src/test/java/com/farmerbb/taskbar/ui/TaskbarShadowScrollView.java

```diff
@@ -0,0 +1,16 @@
+package com.farmerbb.taskbar.ui;
+
+import android.widget.ScrollView;
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.shadows.ShadowScrollView;
+
+@Implements(ScrollView.class)
+public class TaskbarShadowScrollView extends ShadowScrollView {
+
+    @Implementation
+    public void scrollTo(int x, int y) {
+        super.scrollTo(x, y);
```

Comment:
I'm curious to know why this is needed?

Reply:
The `ShadowScrollView` in `robolectric` doesn't shadow `scrollTo` method, and the `scrollTo` invoking will call real `scrollTo` method in `ScrollView`. The `ScrollViwe.scrollTo` will use its child to check whether to scroll:
```java
    @Override
    public void scrollTo(int x, int y) {
        // we rely on the fact the View.scrollBy calls scrollTo.
        if (getChildCount() > 0) {
            View child = getChildAt(0);
            x = clamp(x, getWidth() - mPaddingRight - mPaddingLeft, child.getWidth());
            y = clamp(y, getHeight() - mPaddingBottom - mPaddingTop, child.getHeight());
            if (x != mScrollX || y != mScrollY) {
                super.scrollTo(x, y);
            }
        }
    }
```
If I use the real method, I should setup many parameter for `ScrollView` and its child. But `ShadowView` in `robolectric` provides a shadow method for `scrollTo`, and use its field `scrollToCoordinates` to keep the parameter of `scrollTo`.  So I choose to implement `ShadowScrollView` and add shadow for `scrollTo` to invoke `ShadowView.scrollTo` to shadow the `scrollTo` method to simplify the `scrollTo` test. The `@Implmenetation` annotation is not inherited by subclass, so we should declare the method and add `@Implementation` to it.